libregexp.c

```diff
@@ -2064,7 +2064,7 @@ typedef struct REExecState {
     size_t count; /* only used for RE_EXEC_STATE_GREEDY_QUANT */
     const uint8_t *cptr;
     const uint8_t *pc;
-    void *buf[0];
+    void *buf[];
```

Comment:
We did not use the standard flexible array syntax because it was not supported by older compilers on some targets, possibly in the embedded world.  We do use compound literals so it is unlikely for the compiler to support this and not that. All targets in the CI seem happy so I would OK this change.
An alternative would be to use a macro `FLEXIBLE` defined as nothing on compliant targets and `0` or even `1` on those that do not support the standard syntax.